certbot/cert_manager.py

```diff
@@ -141,6 +142,93 @@ def update_certs_for_domain_matches(candidate_lineage, rv):
 
     return _search_lineages(config, update_certs_for_domain_matches, (None, None))
 
+def cert_path_to_lineage(cli_config):
+    """ If config.cert_path is defined, try to find an appropriate value for config.certname.
+
+    :param .NamespaceConfig cli_config
+    """
+    def archive_files(candidate_lineage, filetype):
+        """ In order to match things like:
+            /etc/letsencrypt/archive/example.com/chain1.pem"""
+        archive_dir = candidate_lineage.archive_dir
+        pattern = [os.path.join(archive_dir, f) for f in os.listdir(archive_dir)
+                        if re.match("{0}[0-9]*.pem".format(filetype), f)]
+        # Using [0] here, so make sure to also check for overlapping archive dirs
+        # if you use this function.
+        return pattern[0]
```

Comment:
it is expected that `len(pattern)` will often equal 0, meaning this will error out.